src/internal/syscall.h

```diff
@@ -185,7 +187,12 @@ static inline long __filter_syscall3(long n, long a1, long a2, long a3) {
 	} else if (n == SYS_msync) {
 		return (long)syscall_SYS_msync((void*)a1, (size_t)a2, (int)a3);
 	} else if (n == SYS_mprotect) {
-		return (long)host_syscall_SYS_mprotect((void*)a1, (size_t)a2, (int)a3);
+		oe_result_t result = sgxlkl_host_syscall_mprotect(&ret_val, (void*)a1, (size_t)a2, (int)a3);
+		if (result != OE_OK) {
+			return -1;
```

Comment:
I wonder if it wouldn't be better to fail when the ocall can't be completed instead of returning an error to the application.